cohttp-async/src/bytebuffer.ml

```diff
@@ -0,0 +1,87 @@
+open Core
```

Comment:
I need to think about how to remove some of this duplication as this is very similar to the module proposed for the lwt backend

Reply:
What are the differences preventing you from making this generic? It seems like you just need a function over a monad + reader.

Reply:
no difference. This can be moved to something that's instantiated using the IO functor (like we do for other modules like request/response)